napari/components/_tests/test_layer_slicer.py

```diff
@@ -244,3 +269,23 @@ def test_slice_layers_async_task_to_layers_lock(layer_slicer):
 
     assert task.result()[layer].id == 1
     assert task not in layer_slicer._layers_to_task
+
+
+def test_slice_layers_async_with_one_3d_image(layer_slicer):
+    np.random.seed(0)
+    data = np.random.rand(8, 7, 6)
+    lockable_data = LockableData(data)
+    layer = Image(data=lockable_data, multiscale=False)
+    dims = Dims(
+        ndim=3,
+        ndisplay=2,
+        range=((0, 8, 1), (0, 7, 1), (0, 6, 1)),
+        current_step=(2, 0, 0),
+    )
+
+    with lockable_data.lock:
```

Comment:
Do you intend on adding a lock to the layers which have async added (as in the FakeAsyncLayer)? If so, seem like it should go in this PR and be added as a test alongside this one (with the layer lock instead of the data lock)

Reply:
No. `FakeAsyncLayer` already has a lock. I did make an update in this PR to pass that lock through to the request, since we now need to acquire it in `__call__`.

But the only reason to add any of these locks is to inject something that allows to control the flow of execution on the slicing thread. So I'm reluctant to test them explicitly.